patches/components-sync-driver-sync_client_mock.h.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/components/sync/driver/sync_client_mock.h b/components/sync/driver/sync_client_mock.h
+index 95005b21493c578f349422f68eb2112a439d8ca4..e3291e0798322d1262c4963dee561a6c6c60a9f3 100644
+--- a/components/sync/driver/sync_client_mock.h
++++ b/components/sync/driver/sync_client_mock.h
+@@ -49,6 +49,10 @@ class SyncClientMock : public SyncClient {
+               (),
+               (override));
+   MOCK_METHOD(void, OnLocalSyncTransportDataCleared, (), (override));
++  MOCK_METHOD(void,
```

Comment:
why are we adding a method to a class with a bunch of associated chromium_src overrides and patches when that method doesn't even access any state in the class? This could be a util  method

Reply:
Because we need dynamic dispatch.

We discussed this with @darkdh in `#sync`.
Currently `BraveSyncServiceImpl` sets the default enabled type to `Bookmarks` here:
https://github.com/brave/brave-core/blob/b71e60c0e4d88dd08ce31083e108fcf5d2d59fe6/components/sync/driver/brave_sync_service_impl.cc#L122-L125
This is what we replace with a call to a `virtual` function:
```
sync_client_->SetDefaultEnabledTypes(this);
```
so that it works for both regular and Rewards sync.

Reply:
How do you tell `BraveSyncServiceImpl` which `ChromeSyncClient` it has to set up the default types for?